test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusIntegrationTestExtension.java

```diff
@@ -194,7 +195,10 @@ private QuarkusTestExtensionState doProcessStart(Properties quarkusArtifactPrope
 
         String artifactType = getArtifactType(quarkusArtifactProperties);
 
-        boolean isDockerLaunch = isContainer(artifactType);
+        Config config = LauncherUtil.installAndGetSomeConfig();
+        String testProfile = TestConfigUtil.integrationTestProfile(config);
+        boolean isDockerLaunch = isContainer(artifactType)
+                || (isJar(artifactType) && "test-with-native-agent".equals(testProfile));
```

Comment:
Is there any way we can be avoid hardcoding the agent here?

Reply:
Hmmm, not sure with the current code base. Docker/container handling code is scattered around in many places. This is another example of surgery we have to keep doing to handle different container build/run setups.

Reply:
Okay, let's keep as is for now and we can figure out something better the next time we hit this